src/methods/clipping/clipping_processor.jl

```diff
@@ -498,20 +562,25 @@ polygons, they are removed from the list
 =#
 function _add_holes_to_polys!(::Type{T}, return_polys, hole_iterator) where T
     n_polys = length(return_polys)
+    remove_poly_idx = fill(false, n_polys)
+    remove_hole_idx = Int[]
     # Remove set of holes from all polygons
     for i in 1:n_polys
         n_new_per_poly = 0
         for curr_hole in hole_iterator # loop through all holes
             # loop through all pieces of original polygon (new pieces added to end of list)
             for j in Iterators.flatten((i:i, (n_polys + 1):(n_polys + n_new_per_poly)))
                 curr_poly = return_polys[j]
-                isnothing(curr_poly) && continue
+                remove_poly_idx[j] && continue
                 n_existing_holes = GI.nhole(curr_poly)
-                curr_poly_ext = n_existing_holes > 0 ? GI.Polygon([GI.getexterior(curr_poly)]) : curr_poly
+                curr_poly_ext = n_existing_holes > 0 ? GI.Polygon(StaticArrays.SVector(GI.getexterior(curr_poly))) : curr_poly
                 in_ext, on_ext, out_ext = _line_polygon_interactions(curr_hole, curr_poly_ext; closed_line = true)
                 if in_ext  # hole is at least partially within the polygon's exterior
-                    new_hole, new_hole_poly, n_new_pieces = _combine_holes!(T, curr_hole, curr_poly, return_polys)
-                    n_new_per_poly += n_new_pieces
+                    new_hole, new_hole_poly, n_new_pieces = _combine_holes!(T, curr_hole, curr_poly, return_polys, remove_hole_idx)
+                    if n_new_pieces > 0
+                        append!(remove_poly_idx, fill(false, n_new_pieces))
```

Comment:
```suggestion
                        append!(remove_poly_idx, falses(n_new_pieces))
```
Takes half the time to materialize the array